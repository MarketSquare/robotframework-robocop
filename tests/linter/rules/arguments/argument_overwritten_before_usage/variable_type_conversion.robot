*** Keywords ***
Move
    [Arguments]    ${distance: int}    ${direction: Literal["UP", "DOWN"]}    ${slow: bool}=True
    VAR    ${distance}     ${10}
    VAR    ${direction}    DOWN
    VAR    ${slow}         ${False}

Move ${distance: int | float} meters
    VAR    ${distance}     ${3.14}
