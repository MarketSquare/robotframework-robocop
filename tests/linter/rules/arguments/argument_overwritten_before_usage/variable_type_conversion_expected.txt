variable_type_conversion.robot:3:20 [W] ARG02 Keyword argument '${distance: int}' is overwritten before usage
variable_type_conversion.robot:3:40 [W] ARG02 Keyword argument '${direction: Literal["UP", "DOWN"]}' is overwritten before usage
variable_type_conversion.robot:3:79 [W] ARG02 Keyword argument '${slow: bool}=True' is overwritten before usage
variable_type_conversion.robot:8:6 [W] ARG02 Keyword argument '${distance: int | float}' is overwritten before usage